### DISCUSSION TITLE
Is there a way to use this project to install llamacpp (when all other methods fail)?

### DISCUSSION THREAD

**Discussion by fullstackwebdev** [withcatai/node-llama-cpp#307] [General]
Great project

I have had a failure on every method of installing llamacpp and it has driven me nuts but this project just "works"

I am wondering, where is it installing the llama cpp and llama cpp python files, is there a way I can copy them or install them globally?

**Comment by giladgd** [comment]
`node-llama-cpp` is not a wrapper around `llama.cpp` binaries but rather a binding, it compiles `llama.cpp` with native functions to expose its functionality on the nodejs side (as well as many high-level functions to make using it as easy as possible).
It ships with prebuilt binaries, so it most likely won't have to compile anything on your machine to work, but when it does, it compiles with CMake, without python.
I've gone to great lengths to ensure there's no python in the process, since configuring python is always cumbersome and not friendly to the extent that leads to frustrations like you probably encountered.
It also includes many workarounds to many compilation issues, to make sure the compilation just works without having to configure anything on the machine for it to work, and in case you need to configure something, it attempts to detect what and tell you exactly what to do.

What do you need `llama.cpp` binaries for that you can't do with the bindings that `node-llama-cpp` provides?
